### PR TITLE
Add `Fail` instance for `Bind ZipList`

### LINKED ISSUE #72: ZipList Bind/Monad
If I understand the new programmable type errors, it seems like this instance can be added:

```purescript
instance zipListIsNotBind
  :: Fail "ZipList is not Bind. Any implementation would break the associativity law"
  => Bind ZipList where
    bind = crashWith "bind: unreachable"
```

Also, does this stop the hierarchy? Or can someone take this `Bind` instance and use it to make a `Monad ZipList` and higher instances?

### PR DESCRIPTION
**Pull request by joneshf**
Closes #72.

### CODE REVIEW

File: src/Data/List/ZipList.purs [hunk 1]
```diff
@@ -65,3 +65,8 @@ instance plusZipList :: Plus ZipList where
   empty = mempty
 
 instance alternativeZipList :: Alternative ZipList
+
+instance zipListIsNotBind
+  :: Fail "ZipList is not Bind. Any implementation would break the associativity law"
```
Comment (paf31) [inline src/Data/List/ZipList.purs]: It might make sense to suggest regular `List` here?

File: src/Data/List/ZipList.purs [hunk 2]
```diff
@@ -65,3 +65,8 @@ instance plusZipList :: Plus ZipList where
...
+  => Bind ZipList where
+    bind _ _ = mempty
```
Comment (paf31) [inline src/Data/List/ZipList.purs]: Can we use `unsafeCrashWith` to avoid confusion (in case anyone thinks this is runnable code)?

**Comment by paf31** [comment]
Thanks!

**Comment by joneshf** [comment]
Updated. Let me know if you'd like the error to be something else. Or if there's precedent somewhere, link me up :smile:

**Comment by garyb** [comment]
New version looks nice! And I think you're setting the precedent here, we've not made use of this anywhere else yet. :smile:

**Comment by paf31** [comment]
👍 LGTM

**Pull request merged**